### DISCUSSION TITLE
require() of ES modules is not supported.

### DISCUSSION THREAD

**Discussion by ivanzwb** [gildas-lormeau/zip.js#365] [Ideas]
When import zip.js in Typescript project.

`import { ZipReader, ZipWriter } from '@zip.js/zip.js';`

Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: \node_modules\@zip.js\zip.js\index.js
require() of ES modules is not supported.

Can we support the commonjs module as well?

**Comment by gildas-lormeau** [comment]
Add `"type": "module"` in your `package.json` file. `zip.js` was not designed for Node.js. It just turned out that as time passed, Node.js became more and more compatible with `zip.js`. That is why it does not offer CommonJS support.

**Comment by peaceps** [comment]
> When import zip.js in Typescript project.
> 
> `import { ZipReader, ZipWriter } from '@zip.js/zip.js';`
> 
> Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: \[node_modules@zip.js](mailto:node_modules@zip.js)\zip.js\index.js require() of ES modules is not supported.
> 
> Can we support the commonjs module as well?

We can use tsimportlib to do async import for commonjs

**Comment by gildas-lormeau** [comment]
I'm moving this issue to the Discussions tab because CommonJS support in zip.js is not planned but we can discuss it if needed.

**Comment by gildas-lormeau** [comment]
The version v2.6.29 adds the support of CommonJS modules via `index.cjs`. You can run for example: 
```js
const { ZipReader, ZipWriter } = require("@zip.js/zip.js/index.cjs");
```
